node.go

```diff
@@ -292,6 +292,9 @@ type groupField struct {
 func (f *groupField) Name() string { return f.name }
 
 func (f *groupField) Value(base reflect.Value) reflect.Value {
+	if base.Kind() == reflect.Interface {
+		base = base.Elem()
+	}
```

Comment:
It'd definitely be beneficial in terms of preventing panics. However, you could consider the provided data in that case to be invalid as the columns would not be marked as optional at that point. I've implemented the nil checks and added some more test cases to illustrate the behaviour. From my perspective this is good enough, but in the commit message I've noted that it might be considered more "correct" to reject the data with a polite error message, but I believe that'd require a refactor.